otherlibs/dune-rpc/private/exported_types.ml

```diff
@@ -34,6 +34,468 @@ module Loc = struct
+      | Fg_bright_cyan -> `Fg_bright_cyan
+      | Fg_bright_white -> `Fg_bright_white
+      | Fg_8_bit_color c -> `Fg_8_bit_color (RGB8.to_ansi_color_rgb8 c)
+      | Fg_24_bit_color c -> `Fg_24_bit_color (RGB24.to_ansi_color_rgb24 c)
+      | Bg_default -> `Bg_default
+      | Bg_black -> `Bg_black
+      | Bg_red -> `Bg_red
+      | Bg_green -> `Bg_green
+      | Bg_yellow -> `Bg_yellow
+      | Bg_blue -> `Bg_blue
+      | Bg_magenta -> `Bg_magenta
+      | Bg_cyan -> `Bg_cyan
+      | Bg_white -> `Bg_white
+      | Bg_bright_black -> `Bg_bright_black
+      | Bg_bright_red -> `Bg_bright_red
+      | Bg_bright_green -> `Bg_bright_green
+      | Bg_bright_yellow -> `Bg_bright_yellow
+      | Bg_bright_blue -> `Bg_bright_blue
+      | Bg_bright_magenta -> `Bg_bright_magenta
+      | Bg_bright_cyan -> `Bg_bright_cyan
+      | Bg_bright_white -> `Bg_bright_white
+      | Bg_8_bit_color c -> `Bg_8_bit_color (RGB8.to_ansi_color_rgb8 c)
+      | Bg_24_bit_color c -> `Bg_24_bit_color (RGB24.to_ansi_color_rgb24 c)
+      | Bold -> `Bold
+      | Dim -> `Dim
+      | Italic -> `Italic
+      | Underline -> `Underline
+    ;;
+
+    let of_ansi_color_style : Stdune.Ansi_color.Style.t -> t = function
+      | `Fg_default -> Fg_default
+      | `Fg_black -> Fg_black
+      | `Fg_red -> Fg_red
+      | `Fg_green -> Fg_green
+      | `Fg_yellow -> Fg_yellow
+      | `Fg_blue -> Fg_blue
+      | `Fg_magenta -> Fg_magenta
+      | `Fg_cyan -> Fg_cyan
+      | `Fg_white -> Fg_white
+      | `Fg_bright_black -> Fg_bright_black
+      | `Fg_bright_red -> Fg_bright_red
+      | `Fg_bright_green -> Fg_bright_green
+      | `Fg_bright_yellow -> Fg_bright_yellow
+      | `Fg_bright_blue -> Fg_bright_blue
+      | `Fg_bright_magenta -> Fg_bright_magenta
+      | `Fg_bright_cyan -> Fg_bright_cyan
+      | `Fg_bright_white -> Fg_bright_white
+      | `Fg_8_bit_color c -> Fg_8_bit_color (RGB8.of_ansi_color_rgb8 c)
+      | `Fg_24_bit_color c -> Fg_24_bit_color (RGB24.of_ansi_color_rgb24 c)
+      | `Bg_default -> Bg_default
+      | `Bg_black -> Bg_black
+      | `Bg_red -> Bg_red
+      | `Bg_green -> Bg_green
+      | `Bg_yellow -> Bg_yellow
+      | `Bg_blue -> Bg_blue
+      | `Bg_magenta -> Bg_magenta
+      | `Bg_cyan -> Bg_cyan
+      | `Bg_white -> Bg_white
+      | `Bg_bright_black -> Bg_bright_black
+      | `Bg_bright_red -> Bg_bright_red
+      | `Bg_bright_green -> Bg_bright_green
+      | `Bg_bright_yellow -> Bg_bright_yellow
+      | `Bg_bright_blue -> Bg_bright_blue
+      | `Bg_bright_magenta -> Bg_bright_magenta
+      | `Bg_bright_cyan -> Bg_bright_cyan
+      | `Bg_bright_white -> Bg_bright_white
+      | `Bg_8_bit_color c -> Bg_8_bit_color (RGB8.of_ansi_color_rgb8 c)
+      | `Bg_24_bit_color c -> Bg_24_bit_color (RGB24.of_ansi_color_rgb24 c)
+      | `Bold -> Bold
+      | `Dim -> Dim
+      | `Italic -> Italic
+      | `Underline -> Underline
+    ;;
+  end
+end
+
+module User_message = struct
+  module Style = struct
+    type t =
```

Comment:
I am unsure if we want to expose the internal details of Dune's error message formatting and make them concrete in the RPC API. @rgrinberg do you think it would be better to make this type opaque and add explicit constructors? It will be difficult to remove constructors from the Stdune version in the future if we keep this around.

Reply:
How would users use the API if this type was completely opaque?